src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs

```diff
@@ -706,11 +706,16 @@ int sizeOfEncodedHeaderAndPayloadAsciiBytes
 
             writer.WriteStartObject();
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
+            if (tokenDescriptor.HasAudience)
             {
                 audienceSet = true;
+                // TODO at next major version Audience will be removed at that time remove this.
                 writer.WritePropertyName(JwtPayloadUtf8Bytes.Aud);
-                writer.WriteStringValue(tokenDescriptor.Audience);
+                writer.WriteStringValue(tokenDescriptor.AudiencesJson);
+/*                if (tokenDescriptor.UseAudiences)
```

Comment:
@keegan-caruso The JsonPrimitives.WriteObject method you recommended was causing encryption test failures that were a bit more than I could quickly figure out, but left this here in case we really would prefer to use that method and want to look into the errors.

Reply:
synced offline, as long as the perf tests don't show a problem this is fine